web3.js/src/connection.ts

```diff
@@ -3154,6 +3176,143 @@ export class Connection {
     return result;
   }
 
+  private async confirmTransactionUsingDurableNonceStrategy({
+    commitment,
+    signature,
+    strategy: {minContextSlot, nonceAccountPubkey, nonceValue},
+  }: {
+    commitment?: Commitment;
+    signature: string;
+    strategy: NonceBasedTransactionConfirmationStrategy;
+  }) {
+    let done: boolean = false;
+    const expiryPromise = new Promise<{
+      __type: TransactionStatus.NONCE_INVALID;
+    }>(resolve => {
+      const INVALID_NONCE_ERROR_SENTINEL = {} as const;
```

Comment:
Why use a sentinel here? Looks like `getCurrentNonceValue` could return `null` if the nonce account wasn't found.

Reply:
This method can throw for reasons other than the account being `null`. I need to be able to disambiguate between ‘you need to break out of this expiry detector by _resolving_ with `NONCE_INVALID`’ vs. ‘you need to throw from this expiry promise because of a fatal.’